packages/starters/backend/sirius-components-starter/src/main/java/org/eclipse/sirius/components/starter/services/EditingContextDispatcher.java

```diff
@@ -42,10 +45,13 @@ public EditingContextDispatcher(IEditingContextEventProcessorRegistry editingCon
 
     @Override
     public Mono<IPayload> dispatchQuery(String editingContextId, IInput input) {
```

Comment:
I'm not sure about asking for a new IDiagramEventProcessor here but I was not able to do it in the editingContextEventProcessor without bringing cyclic dependencies.

Reply:
I did not put the modification on the new architecture on purpose, I wanted to know if it's okay to create the IDiagramEventProcessor just after the invoking the datafetcher.

Reply:
The backend changes should be done in a separate commit, before the one for the frontend (in order to detect changes in the behavior of the code more easily). We should simplify the code of the management of the representation event processors, for that we should start with `IRepresentationEventProcessorFactory` to change it to this:

```
public interface IRepresentationEventProcessorFactory {
    boolean canHandle(IRepresentationConfiguration configuration);

    Optional<IRepresentationEventProcessor> createRepresentationEventProcessor(IRepresentationConfiguration configuration,
            IEditingContext editingContext);
}
```

This way, we don't need the `representationEventProcessorClass` anymore. You can thus remove it from all the representation event processor factories, it should not have an impact since all the factory are using the configuration to find out if they should do their job or not. By removing this class argument from the factory and the composed factory (`IRepresentationEventProcessorComposedFactory`), you can also remove this requirement from `EditingContextEventProcessor#acquireRepresentationEventProcessor`, `IEventProcessorSubscriptionProvider#getSubscription` and all the subscription data fetchers.

Once this is done, we will need to simplify the usage of the `IRepresentationConfiguration`, this will be a bit more tricky. We will probably have to merge things one by one. But first, you will have to remove this class argument otherwise you can't know how to subscribe to a representation using only its identifier.